x-pack/plugins/data_enhanced/server/search/session/check_running_sessions.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  Logger,
+  ElasticsearchClient,
+  SavedObjectsFindResult,
+  SavedObjectsClientContract,
+} from 'kibana/server';
+import {
+  SearchSessionStatus,
+  SearchSessionSavedObjectAttributes,
+  SearchSessionRequestInfo,
+} from '../../../common';
+import { SEARCH_SESSION_TYPE } from '../../saved_objects';
+import { getSearchStatus } from './get_search_status';
+import { getSessionStatus } from './get_session_status';
+import { SearchStatus } from './types';
+
+export async function checkRunningSessions(
+  savedObjectsClient: SavedObjectsClientContract,
+  client: ElasticsearchClient,
+  logger: Logger
+): Promise<void> {
+  try {
+    const runningSearchSessionsResponse = await savedObjectsClient.find<SearchSessionSavedObjectAttributes>(
+      {
+        type: SEARCH_SESSION_TYPE,
+        search: SearchSessionStatus.IN_PROGRESS.toString(),
+        searchFields: ['status'],
+        namespaces: ['*'],
+      }
+    );
+
+    if (!runningSearchSessionsResponse.total) return;
+
+    logger.debug(`Found ${runningSearchSessionsResponse.total} running sessions`);
+
+    const updatedSessions = new Array<SavedObjectsFindResult<SearchSessionSavedObjectAttributes>>();
+
+    let sessionUpdated = false;
+
+    await Promise.all(
+      runningSearchSessionsResponse.saved_objects.map(async (session) => {
+        // Check statuses of all running searches
+        await Promise.all(
+          Object.keys(session.attributes.idMapping).map(async (searchKey: string) => {
+            const updateSearchRequest = (
+              currentStatus: Pick<SearchSessionRequestInfo, 'status' | 'error'>
+            ) => {
+              sessionUpdated = true;
+              session.attributes.idMapping[searchKey] = {
+                ...session.attributes.idMapping[searchKey],
+                ...currentStatus,
+                ...(currentStatus.status === SearchStatus.ERROR ? {} : { error: undefined }),
+              };
+            };
```

Comment:
If the status is 'error', is there an error message available that can be added as a field of `session.attributes`?

Relevant to: https://github.com/elastic/kibana/pull/81707/files#r553623317